pkg/installer/check_test.go

```diff
@@ -0,0 +1,178 @@
+		wantErr bool
+	}{
+		{
+			name:    "empty version, repo as default name",
+			appInfo: "kubernetes-sigs/kustomize",
+			verify: func(o *Installer, version string, t *testing.T) error {
+				assert.Equal(t, version, "latest")
+				assert.Equal(t, o.Org, "kubernetes-sigs")
+				assert.Equal(t, o.Repo, "kustomize")
+				assert.Equal(t, o.Name, "kustomize")
+				return nil
+			},
+			wantErr: false,
+		},
+		{
+			name:    "empty version, specific app name",
+			appInfo: "kubernetes-sigs/kustomize/kustomize",
+			verify: func(o *Installer, version string, t *testing.T) error {
+				assert.Equal(t, version, "latest")
+				assert.Equal(t, o.Org, "kubernetes-sigs")
+				assert.Equal(t, o.Repo, "kustomize")
+				assert.Equal(t, o.Name, "kustomize")
+				return nil
+			},
+			wantErr: false,
+		},
+		{
+			name:    "semver version, specific app name",
+			appInfo: "kubernetes-sigs/kustomize/kustomize@v1.0",
+			verify: func(o *Installer, version string, t *testing.T) error {
+				assert.Equal(t, version, "v1.0")
+				assert.Equal(t, o.Org, "kubernetes-sigs")
+				assert.Equal(t, o.Repo, "kustomize")
+				assert.Equal(t, o.Name, "kustomize")
+				return nil
+			},
+			wantErr: false,
+		},
+		{
+			name:    "specific version with a slash, specific app name",
+			appInfo: "kubernetes-sigs/kustomize/kustomize@kustomize/v1.0",
+			verify: func(o *Installer, version string, t *testing.T) error {
+				assert.Equal(t, version, "kustomize/v1.0")
+				assert.Equal(t, o.Org, "kubernetes-sigs")
+				assert.Equal(t, o.Repo, "kustomize")
+				assert.Equal(t, o.Name, "kustomize")
+				return nil
+			},
+			wantErr: false,
+		},
+		{
+			name:    "specific version with a underlined, specific app name",
+			appInfo: "kubernetes-sigs/kustomize/kustomize@kustomize_v1.0",
+			verify: func(o *Installer, version string, t *testing.T) error {
+				assert.Equal(t, version, "kustomize_v1.0")
+				assert.Equal(t, o.Org, "kubernetes-sigs")
+				assert.Equal(t, o.Repo, "kustomize")
+				assert.Equal(t, o.Name, "kustomize")
+				return nil
+			},
+			wantErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			is := &Installer{}
+			version, err := is.GetVersion(tt.appInfo)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("GetVersion error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if tt.verify != nil {
+				if err := tt.verify(is, version, t); err != nil {
+					t.Errorf("GetVersion() error %v", err)
+				}
+			}
+		})
+	}
+}
+
+func TestProviderURLParseNoConfig(t *testing.T) {
+
+	tests := []struct {
+		name       string
+		packageURL string
+		verify     func(o *Installer, packageURL string, t *testing.T) error
+		wantErr    bool
+	}{
+		{
+			name:       "empty version, repo as default name",
+			packageURL: "orgtest/repotest",
+			verify: func(o *Installer, packageURL string, t *testing.T) error {
+				expectURL := fmt.Sprintf("https://github.com/%s/%s/releases/%s/download/%s-%s-%s.tar.gz",
```

Comment:
I prefer changing it to: `fmt.Sprintf("https://github.com/orgtest/repotest/releases/latest/download/repotest-%s-%s.tar.gz", o.OS, o.Arch)`

Because only `o.OS` and `o.Arch` are varibles.